src/maliput_osm/osm/conversions.cc

```diff
@@ -32,11 +32,86 @@
 #include <algorithm>
 #include <optional>
 #include <string>
+#include <unordered_set>
 
 using maliput_sparse::geometry::LineString3d;
 
 namespace maliput_osm {
 namespace osm {
+namespace {
+
+static constexpr bool kLeftBound{true};
+static constexpr bool kRightBound{!kLeftBound};
+
+// Gets the LaneEnd type of @p connected_ls linestring that connects to @ls linestring.
+// @param ls The line string to get the maliput::api::LaneEnd::Which from.
+// @param connected_ls The line string that shares the initial or end point with ls.
+// @returns The LaneEnd::Which of the connected_ls linestring. If no shared point is found, returns std::nullopt.
+std::optional<maliput::api::LaneEnd::Which> GetLaneEndWhich(const lanelet::ConstLineString3d& ls,
+                                                            const lanelet::ConstLineString3d& connected_ls) {
+  if (ls.front() == connected_ls.front() || ls.back() == connected_ls.front()) {
+    // <--|-->|-->|
```

Comment:
I suggest you explain what <,>, and | are. Otherwise it is somewhat complicated to get the idea of what this is doing.

Reply:
Alternatively, feel free to remove it. The code is self-explanatory.

Reply:
Done.